src/webui/webapplication.cpp

```diff
@@ -265,10 +265,6 @@ void WebApplication::translateDocument(QString &data) const
             // it should fallback to `sourceText`
             QString translation = loadedText.isEmpty() ? sourceText : loadedText;
 
-            // Use HTML code for quotes to prevent issues with JS
```

Comment:
Despite the comment, I haven't found any issues after removing it. I suppose browser console will show errors when the quotes are a problem but that didn't happen when I test.

Reply:
> Despite the comment, I haven't found any issues after removing it. I suppose browser console will show errors when the quotes are a problem but that didn't happen when I test.

Maybe it's just an accident or the incompleteness of the coverage of your tests... I suppose the problem may be if the translation is used as attribute values, e.g. `<tag attr1="QBT_TR(...)QBT_TR">` so after replacing it could be like `<tag attr1="Some text with "quotes" in it">`

Reply:
>Maybe it's just an accident or the incompleteness of the coverage of your tests... 

Maybe.
But what I really mean is, since 5.1 is a new development cycle. We should start try removing this workaround and fix the code properly.

Reply:
> We should start try removing this workaround and fix the code properly.

Quotes can appear in any translated string. Even if you look at all the translation files to make sure that there is no such string exist currently they can appear tomorrow. How it could be "fixed properly"?
>JS code will now try to handle them properly.

What kind of JS code we talk about? And how about the cases when the resulting HTML isn't processed by JS?
> In addition, this workaround actually force JS to use the dangerous `innerHTML` to set the text instead of the safer `textContent`.

Why "this workaround" forces it? Can't HTML code contain such elements (HTML entities etc.) for other reasons?

Reply:
>What kind of JS code we talk about? 

Here is the thread: https://github.com/qbittorrent/qBittorrent/pull/21056#discussion_r1693279666

Reply:
>I haven't found any issues after removing

Same.

Reply:
Okay, continue as you see fit. In any case, replacing only quotes with HTML codes did not look like a general solution.

Reply:
> They should not if the source text doesn't contain quotes.

Ok. But how about the case where source text contains quotes?
Well, source text most likely shouldn't contain them in places where they're not allowed, right? If so, then this can really be solved by generating correct HTML and producing correct translations accordingly.

Reply:
>Well, source text most likely shouldn't contain them in places where they're not allowed, right? If so, then this can really be solved by generating correct HTML and producing correct translations accordingly.

Correct.